### PR TITLE
NLIC-2352 - Add Customer Portal page (wiki)

### PR DESCRIPTION
**Pull request by r-brown**

### CODE REVIEW

**Comment by r-brown** [comment]
Wiki article prepared (see PDF export)
[Customer Portal - NetLicensing Wiki.pdf](https://github.com/Labs64/wiki/files/8034862/Customer.Portal.-.NetLicensing.Wiki.pdf)

File: docs/customer-portal.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      <ns2:property name="action">licenseeLogin</ns2:property>
+      <ns2:property name="licenseeNumber">I001-DEMO</ns2:property>
+      <ns2:property name="vendorNumber">VDEMO</ns2:property>
+    </ns2:item>
+  </ns2:items>
+</ns2:netlicensing>
+```
+{: .ml-5 }
+
+The value of the attribute `customerPortalURL` in the successful token create response need to be used to forward customer to hosted Customer Portal page.
```
Comment (kkorotkov) [inline docs/customer-portal.md]: Proposal: `Use the 'customerPortalURL' value from the above response to redirect the customer to the Customer Portal.`

File: docs/restful-api/services/token-services.md [hunk 2]
```diff
@@ -228,19 +228,19 @@ Content-Type: application/x-www-form-urlencoded</p></td>
 - <i>DEFAULT</i> - default one-time token (will be expired after first request)<br />
 - <i>SHOP</i> - shop token is used to redirect customer to the NetLicensing Shop ('licenseeNumber' is mandatory)<br />
 - <i>APIKEY</i> - APIKey token<br/>
-- <i>ACTION</i> - Action token (e.g. Customer Portal)<br/>
+- <i>ACTION</i> - Action token (e.g. <a href="customer-portal">Customer Portal</a>)<br/>
```
Comment (kkorotkov) [inline docs/restful-api/services/token-services.md]: Proposal: `(e.g. access token for the <a href="customer-portal">Customer Portal</a>)`

File: docs/restful-api/services/token-services.md [hunk 3]
```diff
@@ -228,19 +228,19 @@ Content-Type: application/x-www-form-urlencoded</p></td>
...
 </p>
 <p>
 licenseeNumber (string) -
 <br>- <i>tokenType=SHOP</i> - identifies licensee that will be assigned to the shop token
 <br>- <i>tokenType=APIKEY</i> - specifies the scope of validity of the APIKey; the only licensee with the given number can be validated using this APIKey</p>
+<p>action (enum) - 'licenseeLogin'; only for token=ACTION.</p>
+<p>licenseeNumber (string) - customer number; only for token=ACTION.</p>
```
Comment (kkorotkov) [inline docs/restful-api/services/token-services.md]: `licenseeNumber` is documented a few lines above for various token types.

File: docs/customer-portal.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+Customer Portal is a white-label solution and can be adjusted accordingly to your brand identity.
+
+**Adjustable elements:**
+
+- **Logo** - Vendor logo can be loaded at "Settings / Shop Customization". This logo will be shown in the Customer Portal header and open vendor card on click.
+- **Accent colour** - Accent colour can be specified at "Settings / Shop Customization" as "Primary Color".
+- **Vendor card** - Vendor card is opened on logo click and contains vendor data as it s filled at "Profile"
+- **Legal information** - The valid URLs for the "Terms of Service", "Privacy Policy" and "Imprint" hyperlinks in the footer are being taken from "Profile / Legal Information (tab)".
+- **Product information** - Product and product module details (such as Name, Description and Logo) can be provided for each configured product in the system.
```
Comment (kkorotkov) [inline docs/customer-portal.md]: Screenshots?

File: docs/customer-portal.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  </ns2:items>
+</ns2:netlicensing>
+```
+{: .ml-5 }
+
+The value of the attribute `customerPortalURL` in the successful token create response need to be used to forward customer to hosted Customer Portal page.
+
+### Customization & Branding Options
+
+Customer Portal is a white-label solution and can be adjusted accordingly to your brand identity.
```
Comment (kkorotkov) [inline docs/customer-portal.md:65]: Add a sentence, explaining where to find the adjustable elements, e.g.: `The following elements can be adjusted via the NetLicensing UI:`

**Review by kkorotkov — CHANGES_REQUESTED** [review]

File: docs/customer-portal.md (hunk 1 above)
Comment (r-brown) [reply docs/customer-portal.md]: changed

File: docs/restful-api/services/token-services.md (hunk 2 above)
Comment (r-brown) [reply docs/restful-api/services/token-services.md]: changed

File: docs/customer-portal.md (hunk 4 above)
Comment (r-brown) [reply docs/customer-portal.md]: screenshots added

File: docs/customer-portal.md (hunk 5 above)
Comment (r-brown) [reply docs/customer-portal.md:65]: changed

File: docs/restful-api/services/token-services.md (hunk 3 above)
Comment (r-brown) [reply docs/restful-api/services/token-services.md]: fixed

File: docs/restful-api/services/token-services.md [hunk 6]
```diff
@@ -228,14 +228,13 @@ Content-Type: application/x-www-form-urlencoded</p></td>
...
 - <i>SHOP</i> - shop token is used to redirect customer to the NetLicensing Shop ('licenseeNumber' is mandatory)<br />
 - <i>APIKEY</i> - APIKey token<br/>
-- <i>ACTION</i> - Action token (e.g. <a href="customer-portal">Customer Portal</a>)<br/>
+- <i>ACTION</i> - Action token (e.g. access token for the <a href="customer-portal">Customer Portal</a>)<br/>
 </p>
 <p>
-licenseeNumber (string) -
+licenseeNumber (string) - unique customer number
 <br>- <i>tokenType=SHOP</i> - identifies licensee that will be assigned to the shop token
 <br>- <i>tokenType=APIKEY</i> - specifies the scope of validity of the APIKey; the only licensee with the given number can be validated using this APIKey</p>
```
Comment (kkorotkov) [inline docs/restful-api/services/token-services.md]: Explanation for tokenType=ACTION is missing.

**Review by kkorotkov — CHANGES_REQUESTED** [review]

File: docs/restful-api/services/token-services.md (hunk 6 above)
Comment (r-brown) [reply docs/restful-api/services/token-services.md]: fixed

**Review by kkorotkov — APPROVED** [review]

**Pull request merged**